### ISSUE TITLE
Unable to delete realm (In Unit Test) on Upgrade to 1.60 from 0.81

### ISSUE THREAD

**Issue by sisterray** [realm/realm-dotnet#1545]
## Goals

Successfully run a test that passed in v0.81 but does not in v1.6

## Expected Results

The test should pass

## Actual Results
The test fails with Realms.Exceptions.RealmPermissionDeniedException. Unable to delete Realm because it is still open

## Steps to Reproduce

I am running this test using a Xamarin Android Unit Test library which references a shared project

## Code Sample

        [Test]
        public void RealmPotentialBugTest()
        {
            // This is a stripped down test for repeating potential realm bug
            var config1 = new RealmConfiguration("Dummy.Realm");
            // Generate encryption key for test
            var rand = new Random();
            byte[] encryptionKey = new byte[64];
            rand.NextBytes(encryptionKey);
            config1.EncryptionKey = encryptionKey;

            var realm = Realm.GetInstance(config1);

            // This is a bit fake normally this code is in the [TearDown] method
            var config2 = new RealmConfiguration("Dummy.Realm");
            config2.EncryptionKey = encryptionKey;
            Realm.DeleteRealm(config2);
        }

## Version of Realm and Tooling

- Realm Object Server Version: N/A
- Flavor:
  - [ ] Developer
  - [ ] Professional
  - [ ] Enterprise
- Server OS & Version: ? (e.g. CentOS 6)
- Client SDK Version: ?
- Client OS & Version: Android 4.4.4
Realm dotnet DB version 1.6.0

**Comment by sisterray** [comment]
Just thought what this could be, so I changed the code to run the delete section in a new thread and it worked

`   
    

     [Test]
    public void RealmPotentialBugTest()
    {
        // This is a stripped down test for repeating potential realm bug
        var config1 = new RealmConfiguration("Dummy.Realm");
        // Generate encryption key for test
        var rand = new Random();
        byte[] encryptionKey = new byte[64];
        rand.NextBytes(encryptionKey);
        config1.EncryptionKey = encryptionKey;var realm = Realm.GetInstance(config1);

       var task = Task.Run(() =>
        {
             // This is a bit fake normally this code is in the [TearDown] method
             var config2 = new RealmConfiguration("Dummy.Realm");
             config2.EncryptionKey = encryptionKey;
             Realm.DeleteRealm(config2);
        });
        task.Wait(5000);
    }`

Was this a fix made since v0.81? Could someone from the team confirm that this is the expected behaviour

**Comment by fealebenpae** [comment]
Instead of deleting in a worker thread, try disposing the realm before deleting it - the exception says that you can’t delete the realm file because it’s opened by the `var realm = ...` statement.

**Comment by nirinchev** [comment]
And yes, this was a fix made after 0.81.0 and is intended behavior - realms that are currently open cannot be safely deleted, so as @fealebenpae suggested, wrap the realm in a using:
```csharp
var config = ...;
using (var realm = Realm.GetInstance(config))
{
    // Do your test
}
Realm.DeleteRealm(config);
```

**Comment by sisterray** [comment]
OK, Thanks for clearing that up. I will close this issue.
